includes/Checker/Check_Categories.php

```diff
@@ -0,0 +1,58 @@
+<?php
+/**
+ * Class WordPress\Plugin_Check\Checker\Check_Categories
+ *
+ * @package plugin-check
+ */
+
+namespace WordPress\Plugin_Check\Checker;
+
+/**
+ * Abstract Check Runner class.
+ *
+ * @since n.e.x.t
+ */
+class Check_Categories {
```

Comment:
Can you add PHPUnit test coverage for this class?

For `get_categories()`, you can probably use a simple check based on https://www.php.net/manual/en/reflectionclass.getconstants.php to ensure the returned array includes all the `CATEGORY_*` class constants.

For `filter_checks_by_categories()`, you can maybe use a data provider to cover multiple scenarios and ensure the expected checks are filtered based on the categories passed.

Reply:
We could also just update `get_categories()` to return all class constants. Something like:

```php
public function get_categories() {
		$consts = ( new ReflectionClass( $this ) )->getConstants();

		/**
		 * List of categories.
		 *
		 * @var string[] $categories
		 */
		$categories = array_values(
			array_filter(
				$consts,
				static fn( $key ) => str_starts_with( $key, 'CATEGORY_' ),
				ARRAY_FILTER_USE_KEY
			)
		);

		return $categories;
	}

Reply:
Good idea. In that case maybe we should "cache" that in a static variable though to avoid recomputing this every time `get_categories()` is called?

Reply:
Thank you both for the feedback. In https://github.com/10up/plugin-check/pull/220/commits/caa2247b5d254c860b65b47d7b65f7bb3438b667 and https://github.com/10up/plugin-check/pull/220/commits/a5a67c7153ebb43f5d82203e6b810a71446a3b0f i have address the review feedback and add unit tests.